android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdk.java

```diff
@@ -1,11 +1,83 @@
 package com.hypertrack.sdk.capacitor;
 
 import android.util.Log;
+import com.hypertrack.sdk.HyperTrack;
+import com.hypertrack.sdk.TrackingError;
+import com.hypertrack.sdk.TrackingStateObserver;
 
-public class HyperTrackSdk {
+public class HyperTrackSdk  implements TrackingStateObserver.OnTrackingStateChangeListener{
 
-    public String echo(String value) {
-        Log.i("Echo", value);
+    private static final String TAG = "HyperTrackPlugin";
+    private HyperTrack sdkInstance;
+
+    public String print(String value) {
+        Log.i(TAG, value);
         return value;
     }
-}
+
+    @Override
+    public void onError(TrackingError trackingError) {
+        //TO DO
+    }
+
+    @Override
+    public void onTrackingStart() {
+        print("Tracking is started");
+    }
+
+    @Override
+    public void onTrackingStop() {
+        print("Tracking is stopped");
+    }
+
+    public void Stop() {
+        if (sdkInstance == null) {
+            throw new IllegalStateException("Sdk wasn't initialized");
+        }
+        if (!sdkInstance.isTracking()) {
+            throw new IllegalStateException("Tracking wasn't started");
+        }
+        try {
+            sdkInstance.stop();
+            return;
+        } catch (Exception e) {
+            throw new IllegalArgumentException(e.getMessage());
+        }
+
+    }
+
+    public void Start() {
+        if (sdkInstance == null) {
+            throw new IllegalStateException("Sdk wasn't initialized");
+        }
+        try {
+            sdkInstance.start();
+            return;
+        } catch (Exception e) {
+            throw new IllegalArgumentException(e.getMessage());
```

Comment:
this way you will lose the stacktrace, please return the original exception, or add it as a cause:

new IllegalArgumentException(e)

Reply:
please fix it in other places too